cmd/scollector/collectors/disk_linux.go

```diff
@@ -169,7 +169,10 @@ func examineMdadmVolume(volumeName string) (volumeDetail, error) {
 	// We don't use --test because it has failed us in the past.
 	// Maybe we should use it sometime in the future
 	output, err := util.Command(tmout, nil, "mdadm", "--detail", volumeName)
-	detail := parseExamineMdadm(output)
+	detail := volumeDetail{}
```

Comment:
Just a minor thing. I'm much more used to seeing `if err != nil` blocks. How about:

```
if err != nil{
   return volumeDetail{}, err
}
return parseExamineMdadm(output), nil
```